### PR TITLE
Tweaked Intimidate a little

### PR DESCRIPTION
**Pull request by LOuroboros**
# Description
This PR addresses one of the many mechanical differences posted by @ExpoSeed in #1215 
I made Intimidate affect opponents with Inner Focus, Scrappy, Oblivious or Own Tempo based on a config.
I also added a new text string for the ability, one that is much closer to the one used by the Gen. 8 Games.
I didn't put it behind a config because it felt a bit too much to me, but if it's really needed, I'll give that a shot.

## **Discord contact info**
Lunos#4026

## Misc.

It's worth to note that the Ability PopUp on the Player's side changes to the Opponent's ability for a split second.
Someone else is taking care of that in a separate PR as it seems to be more directly related to that, than it is to Intimidate.

Also, I didn't remove the original string used by the ability *(`STRINGID_PREVENTEDFROMWORKING`)* just in case.

If there's anything wrong with this PR, let me know.

### CODE REVIEW

**Review by ghoulslash — APPROVED** [review]

File: include/constants/battle_config.h [hunk 1]
```diff
@@ -106,6 +106,7 @@
 #define B_MOODY_ACC_EVASION         GEN_6 // In Gen8+, Moody CANNOT raise Accuray and Evasion any more.
 #define B_FLASH_FIRE_FROZEN         GEN_6 // In Gen5+, Flash Fire can trigger even when frozen, when it couldn't before.
 #define B_SYNCHRONIZE_NATURE        GEN_6 // In Gen8+, if the Pokémon with Synchronize is leading the party, it's 100% guaranteed that wild Pokémon will have the same ability, as opposed to 50% previously.
+#define B_EXPANDED_INTIMIDATE       GEN_8 // In Gen8, Intimidate doesn't work on opponents with the Inner Focus, Scrappy, Own Tempo or Oblivious abilities.
```
Comment (ExpoSeed) [inline include/constants/battle_config.h:109]: i'm not sure if `B_EXPANDED_INTIMIDATE` is the best name given that what happened to intimidate was the opposite of an expansion.

would something like `B_INTIMIDATE_BLOCK` be better maybe, or is it fine as is?

Comment (LOuroboros) [reply include/constants/battle_config.h:109]: `B_INTIMIDATE_BLOCK` would be more appropriate if Intimidate didn't already block certain abilities by default _(and it does, more specifically Hyper Cutter, Clear Body and White Smoke)_.

My reasoning behind `B_EXPANDED_INTIMIDATE` is that here, I basically expanded the ability's reach in terms of which opposing abilities can block it from activating.
I'm opened to suggestions if that's no good though.

Comment (ExpoSeed) [reply include/constants/battle_config.h:109]: As discussed on Discord, I like Lunos's suggestion of `B_UPDATED_INTIMIDATE` most.

Comment (LOuroboros) [reply include/constants/battle_config.h:109]: Yeah, me too. I think he got a good idea this time around. I'll probably stick to it. His label reflects the effect of the configuration better.

**Review by ExpoSeed — APPROVED** [review]

**Review by ghoulslash — APPROVED** [review]

**Pull request merged**